### PR TITLE
CLDR-14251 Modernize: eliminate ajax_status.jsp

### PR DESCRIPTION
**Pull request by btangmu**
-Move most of ajax_status.jsp to SurveyAjax.includeJavaScript, converting jsp to java

-New CldrStatus.js to encapsulate data like surveyRunningStamp on front end with getters/setters

-TEMPORARY SurveyAjax.includeEvilJavaScript for items not yet handled by CldrStatus.js

-Extend statusJSON to include contextPath

-Remove unused front-end variables surveyCurrentLocaleStamp, surveyCurrentLocaleStampId, etc.

-Remove dead code; clean up

-Comments

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-14251
- [x] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

**Comment by btangmu** [comment]
I'm going ahead with this PR even though there's lots more to be done and these transitional changes are a bit ugly. It seems best to have small incremental PRs than change too much in a single PR. I'll follow up soon with eliminating v.jsp, and also replacing includeEvilJavaScript() with something less evil.

File: tools/cldr-apps/src/main/webapp/js/special/users.js [hunk 1]
```diff
@@ -1,204 +1,202 @@
-/**
- * Special module that shows a users page.
```
Comment (btangmu) [inline tools/cldr-apps/src/main/webapp/js/special/users.js:2]: All the whitespace changed in this file, but the only substantial difference is contextPath changed to cldrStatus.getContextPath()

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java [hunk 2]
```diff
@@ -3185,47 +3173,177 @@ public String getValue() {
...
+            out.write("var surveyUserPerms = {userExist: false,};\n");
+        }
+        out.write("var warnIcon = \"" + WebContext.iconHtml(request, "warn", "Test Warning") + "\";\n");
+        out.write("var stopIcon = \"" + WebContext.iconHtml(request, "stop", "Test Error") + "\";\n");
+        out.write("var WHAT_GETROW = '" + SurveyAjax.WHAT_GETROW + "';\n");
+        out.write("var WHAT_SUBMIT = '" + SurveyAjax.WHAT_SUBMIT + "';\n");
+        out.write("var TARGET_DOCS = '" + WebContext.TARGET_DOCS + "';\n");
+        out.write("var TRANS_HINT_LOCALE = '" + SurveyMain.TRANS_HINT_LOCALE + "';\n");
+        out.write("var TRANS_HINT_LANGUAGE_NAME = '" + SurveyMain.TRANS_HINT_LANGUAGE_NAME + "';\n");
+        out.write("var WHAT_GETROW = '" + SurveyAjax.WHAT_GETROW + "';\n");
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java:3308]: Oops WHAT_GETROW is accidentally duplicated here. It's harmless but I'll remove it in the next PR.

Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java:3308]: it's somewhat temporary anyway…

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java [hunk 3]
```diff
@@ -384,14 +384,14 @@ private void processRequest(HttpServletRequest request, HttpServletResponse resp
                 addGeneralStats(r);
                 send(r, out);
             } else if (what.equals(WHAT_FLAGGED)) {
-                JSONWriter r = newJSONStatus(sm);
+                JSONWriter r = newJSONStatus(request, sm);
```
Comment (srl295) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java:387]: I wonder if these `r` instances should be hoisted up, since they are almost always used.  BUt, not really important.

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java (hunk 2 above)
Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java:3308]: these WHAT_ things shouldn't need to be communicated to the client… hard coded would be fine. But no reason to change it now.

File: tools/cldr-apps/src/main/webapp/v.jsp [hunk 4]
```diff
@@ -203,12 +213,18 @@ if(false) { // if we need to redirect for some reason..
...
 <link rel="stylesheet" href="<%=request.getContextPath()%>/surveytool.css" />
 <link rel="stylesheet" href="<%=request.getContextPath()%>/css/CldrStForum.css" />
-<%
-	// TODO: when v.jsp includes ajax_status.jsp, avoid redundant links for bootstrap, surveytool.css, redesign.css
-	// Note that ajax_status.jsp now reliably should output the surveyUser, so we don't need
-	// to output it again here.
-%>
-<%@include file="/WEB-INF/tmpl/ajax_status.jsp" %>
+
+<link rel="stylesheet" href="//ajax.googleapis.com/ajax/libs/dojo/1.14.1/dijit/themes/claro/claro.css" />
```
Comment (srl295) [inline tools/cldr-apps/src/main/webapp/v.jsp:217]: why is this added here?

Comment (srl295) [reply tools/cldr-apps/src/main/webapp/v.jsp:217]: oh, this is an old fix… thanks!

File: tools/cldr-apps/src/main/webapp/surveytool.css [hunk 5]
```diff
@@ -8,40 +8,14 @@
 /* Made with CSSEdit and Espresso2 */
 @charset "UTF-8";
 
-
 /* lang specific CSS */
 
-/*
-NOTE: See http://www.w3.org/International/questions/qa-css-lang
```
Comment (srl295) [inline tools/cldr-apps/src/main/webapp/surveytool.css:15]: These are instructions to someone maintaining this as to how to add a new webfont, such as done for Myanmar

**Review by srl295 — APPROVED** [review]
good stuff! minor comments, and part of the .css that could be kept, but great progress

File: tools/cldr-apps/src/main/webapp/surveytool.css (hunk 5 above)
Comment (btangmu) [reply tools/cldr-apps/src/main/webapp/surveytool.css:15]: I'll restore the comment.

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java (hunk 3 above)
Comment (btangmu) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java:387]: I agree it's too much duplication. processRequest is over 700 lines, so nontrivial to refactor. When we do, good idea to consolidate calls to newJSONStatus
